### PR TITLE
Enable CheckJavaStyle presubmit check

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Resolves https://github.com/brave/brave-browser/issues/30823

This PR does following:
1. Enables `CheckJavaStyle` checks for Brave java files
2. Introduces `brave-style-5.0.xml` file which is the copy of `chromium-style-5.0.xml`, where severity for all checks except RedundantImport/UnusedImports is changed error => warning
3. Fixes RedundantImport/UnusedImports style error in all files.

Remaining style errors are here attached: [presubmit_result.txt](https://github.com/brave/brave-core/files/11666460/presubmit_result.txt)

It is suggested to make review per commit, only 1st has important change; the rest are generated with script.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
I. presubmit errors
1. Create PR with some java file changed
2. Add unused import
3. Commit change, run `npm run presubmit` - expected to see error about unused import

II. presubmit warnings 
1. Create PR with some java file changed
2. Add method 
```Java
private int GetWidth() {
    return 0;
}
```
3. Commit change, run `npm run presubmit` - expected to see warning `Method names should start with a lower case letter (e.g. getWidth())`

### CODE REVIEW

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Review by Pavneet-Sing — APPROVED** [review]

**Review by tmancey — APPROVED** [review]
++

**Comment by deeppandya** [comment]
Thank you @AlexeyBarabash LGTM

**Review by deeppandya — APPROVED** [review]

**Review by simoarpe — APPROVED** [review]
👍 LGTM

**Review by goodov — APPROVED** [review]
lgtm witth few nits

**Review by samartnik — APPROVED** [review]
java ++

File: PRESUBMIT.py [hunk 1]
```diff
@@ -343,4 +343,76 @@ def _ChangeHasSecurityReviewer(*_):
     return False
 
 
+@chromium_presubmit_overrides.override_check(globals())
+# pylint: disable=unused-argument
+def CheckJavaStyle(original_check, input_api, output_api):
```
[the remark this answers is not in the document]
Comment (AlexeyBarabash) [reply PRESUBMIT.py]: thanks, fixed at 
https://github.com/brave/brave-core/compare/50b8e3289b63d75ad8dcea2a799438278e79e23e..5fc7576cf25caf430aa4aac2c4f5bbb891d30bdc#diff-44060d1ca29be9730234254fd446bf8b450c079ef1d23084623124820c41e9aaR347

File: PRESUBMIT.py [hunk 2]
```diff
@@ -343,4 +343,76 @@ def _ChangeHasSecurityReviewer(*_):
...
+    import sys
+    original_sys_path = sys.path
+    try:
+        sys.path = sys.path + [
+            input_api.os_path.join(input_api.PresubmitLocalPath(), 'tools',
+                                   'android', 'checkstyle')
+        ]
+        # pylint: disable=import-error
+        import checkstyle
+        # pylint: enable=import-error
```
[the remark this answers is not in the document]
Comment (AlexeyBarabash) [reply PRESUBMIT.py]: thanks, fixed at 
https://github.com/brave/brave-core/compare/50b8e3289b63d75ad8dcea2a799438278e79e23e..5fc7576cf25caf430aa4aac2c4f5bbb891d30bdc#diff-44060d1ca29be9730234254fd446bf8b450c079ef1d23084623124820c41e9aaR362
and 
https://github.com/brave/brave-core/compare/50b8e3289b63d75ad8dcea2a799438278e79e23e..5fc7576cf25caf430aa4aac2c4f5bbb891d30bdc#diff-44060d1ca29be9730234254fd446bf8b450c079ef1d23084623124820c41e9aaR370

It  turns out that in a previous version I disabled that checks for all file 
(https://pylint.readthedocs.io/en/latest/user_guide/messages/message_control.html)

File: PRESUBMIT.py [hunk 3]
```diff
@@ -343,4 +343,76 @@ def _ChangeHasSecurityReviewer(*_):
...
+    if not any(
+            _IsJavaFile(input_api, f.LocalPath())
+            for f in input_api.AffectedFiles()):
+        return []
+
+    # pylint: disable=import-outside-toplevel
+    import sys
+    original_sys_path = sys.path
+    try:
+        sys.path = sys.path + [
```
[the remark this answers is not in the document]
Comment (AlexeyBarabash) [reply PRESUBMIT.py]: thanks, fixed at https://github.com/brave/brave-core/compare/50b8e3289b63d75ad8dcea2a799438278e79e23e..5fc7576cf25caf430aa4aac2c4f5bbb891d30bdc#diff-44060d1ca29be9730234254fd446bf8b450c079ef1d23084623124820c41e9aaR367

**Pull request merged**